pkg/client/client.go

```diff
@@ -193,79 +182,12 @@ func GetKubeConfigFile() string {
 	return filename
 }
 
-//restScope implementation
-type restScope struct {
-	name apimeta.RESTScopeName
-}
-
-func (r *restScope) Name() apimeta.RESTScopeName {
-	return r.name
-}
-
 // newControllerCliOptions creates the mapper and schema options for the inner fallback cli. If set to defaults, the Controller Cli will try
 // to discover the mapper by itself by querying the API, which can take too much time. Here we're setting this mapper manually.
 // So it's need to keep adding them or find some kind of auto register in the kube api/apimachinery
-func newControllerCliOptions() controllercli.Options {
-	options := controllercli.Options{}
-	builder := meta.GetRegisteredSchemeBuilder()
-	gvks, err := getGVKsFromAddToScheme(builder.AddToScheme)
-	if err != nil {
-		log.Error(err, "Error while creating SchemeBuilder for Kubernetes client")
-		panic(err)
-	}
-	mapper := apimeta.NewDefaultRESTMapper([]schema.GroupVersion{})
-	for _, gvk := range gvks {
-		// namespaced resources
-		if (gvk.GroupVersion() == corev1.SchemeGroupVersion && gvk.Kind == "Namespace") ||
-			(gvk.GroupVersion() == apiextensionsv1beta1.SchemeGroupVersion && gvk.Kind == "CustomResourceDefinition") ||
-			(gvk.GroupVersion() == rbac.SchemeGroupVersion && gvk.Kind == "ClusterRole") ||
-			(gvk.GroupVersion() == rbac.SchemeGroupVersion && gvk.Kind == "ClusterRoleBinding") {
-			mapper.Add(gvk, &restScope{name: apimeta.RESTScopeNameRoot})
-		} else { // everything else
-			mapper.Add(gvk, &restScope{name: apimeta.RESTScopeNameNamespace})
-		}
+func newControllerCliOptions(config *rest.Config) controllercli.Options {
+	options := controllercli.Options{
```

Comment:
mapper use the dynamic one: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L68